### PR TITLE
Plants index endpoint

### PR DESCRIPTION
**Pull request by marizaswart**
## Description
Creating an API with an endpoint for plants to enable the frontend to consume the API endpoint and display a list of plants to the users.

## What we have learnt
To set up an API only rails application with unit tests

## Screenshot
API tested with postman
screen shot 2018-01-28 at 17 23 09

### CODE REVIEW

File: spec/models/plant_spec.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+    it { is_expected.to have_db_column :id }
+    it { is_expected.to have_db_column :name }
+    it { is_expected.to have_db_column :description }
+  end
+
+  describe 'Validations' do
+    it { is_expected.to validate_presence_of :name }
+    it { is_expected.to validate_presence_of :description }
+  end
+end
```
Comment (tochman) [inline spec/models/plant_spec.rb]: I would add a spec for the factory

**Review by tochman — COMMENTED** [review]
Looks okay imo. I would look into how to test serializers. If you want I could take a shot at it. Alos, I would set up Coveralls for the final project. It feels like a good thing to be able to say. "we have xx% coverage!"

File: spec/models/plant_spec.rb (hunk 1 above)
Comment (marizaswart) [reply spec/models/plant_spec.rb]: Working on it now, missed it

**Comment by marizaswart** [comment]
We have coveralls, just need to fix it so it will show: https://coveralls.io/builds/15258150
Coverage is low, working on more specs

**Comment by tochman** [comment]
I see. Cool. O would remove job, channel and mailer folders/files for now. Thes files are not needed atm. That will boost your test coverage.

**Pull request merged**